manta-trusted-setup/src/groth16/ppot/serialization.rs

```diff
@@ -610,7 +609,8 @@ where
         Compressed::No => cfg_iter!(powers).for_each(|g| curve_point_checks(g).unwrap()),
         Compressed::Yes => cfg_iter!(powers).for_each(|g| {
             if !g.is_in_correct_subgroup_assuming_on_curve() {
```

Comment:
question here: How do you check whether a point belongs to a subgroup without a discrete log?

Reply:
Right, in principle you'd need a discrete log to prove that an arbitrary group element belongs to a particular cyclic subgroup. But if that subgroup has prime order `r` and is the only subgroup of order `r` then it is sufficient to check that `r * h = 0` to prove that `h` belongs to the given subgroup.  That's what this check does, it multiplies the point by `r` and checks that the result is 0.

Reply:
Now that I think more about it I suppose there's no need to assume `r` is prime in what I wrote above, it's enough for the group to have only one subgroup of order `r`. (Though in our case `r` is indeed prime)